plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -95,82 +105,185 @@ public ConnectorTableMetadata getTableMetadata()
     }
 
     @Override
-    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
     {
+        List<io.trino.spi.type.Type> types = getType(tableMetadata);
+        Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
```

Comment:
I'd move these down and use `InMemoryRecordSet.cursor()` instead of `PlanFilesIterable` so that you don't need these variables in the empty snapshotId case

```
        if (snapshotId.isEmpty()) {
            return new InMemoryRecordSet(types, ImmutableList.of()).cursor();
        }

        List<io.trino.spi.type.Type> types = getType(tableMetadata);
        Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
```